integration/common.go

```diff
@@ -0,0 +1,120 @@
+// +build linux
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package integration
+
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	yaml "gopkg.in/yaml.v2"
+	cri "k8s.io/cri-api/pkg/apis"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+// RegistryList holds public image registries
+type RegistryList struct {
+	DockerLibraryRegistry    string `yaml:"dockerLibraryRegistry"`
+	K8sCriContainerdRegistry string `yaml:"k8sCriContainerdRegistry"`
+	K8sGcrRegistry           string `yaml:"k8sGcrRegistry"`
+}
+
+// Config holds an images registry, name, and version
+type Config struct {
+	registry string
+	name     string
+	version  string
+}
+
+var (
+	imageService             cri.ImageManagerService
+	imageConfigs             map[int]Config
+	dockerLibraryRegistry    string
+	k8sCriContainerdRegistry string
+	k8sGcrRegistry           string
+	pauseImage               string // This is the same with default sandbox image
+)
+
+func initReg(repoList string) {
+	registry := RegistryList{
+		DockerLibraryRegistry:    "docker.io/library",
+		K8sCriContainerdRegistry: "gcr.io/k8s-cri-containerd",
+		K8sGcrRegistry:           "k8s.gcr.io",
+	}
+
+	if repoList != "" {
+		fileContent, err := ioutil.ReadFile(repoList)
+		if err != nil {
+			panic(fmt.Errorf("Error reading '%v' file contents: %v", repoList, err))
+		}
+
+		err = yaml.Unmarshal(fileContent, &registry)
+		if err != nil {
+			panic(fmt.Errorf("Error unmarshalling '%v' YAML file: %v", repoList, err))
+		}
+	}
+
+	dockerLibraryRegistry = registry.DockerLibraryRegistry
+	k8sCriContainerdRegistry = registry.K8sCriContainerdRegistry
+	k8sGcrRegistry = registry.K8sGcrRegistry
+
+	imageConfigs = initImageConfigs()
+	pauseImage = GetImage(Pause)
+}
+
+const (
+	// None is to be used for unset/default images
+	None = iota
+	// BusyBox image
+	BusyBox
+	// Pause image
+	Pause
+	// VolumeCopyUp image
+	VolumeCopyUp
+	// VolumeOwnership image
+	VolumeOwnership
+)
+
+func initImageConfigs() map[int]Config {
+	configs := map[int]Config{}
+	configs[BusyBox] = Config{dockerLibraryRegistry, "busybox", "1.29"}
```

Comment:
Why not to just specify image URLs like `docker.io/library/busybox:1.29` ?
Something like a KV map:

```
map[string]string {
   { "busybox": "docker.io/library/busybox:1.29" },
   { "pause": "" },
...
}

GetImage("busybox") -> string
```

This way you don't need to compose URL and overall logic would be much simpler to maintain/add new images.

Reply:
Done

Reply:
Is there a reason why we want to specify registry separately and than concat URL with container image name?
I was thinking about something simple like dumb txt file like:
```
busybox=docker.io/library/busybox:1.32
nginx=...
pause=...
```

And in integration test we can call GetImage("busybox") -> "docker.io/library/busybox:1.32"

Would this make sense?
Apologies if I didn't explain my point well and confused you.

Reply:
Done